### PR TITLE
DR-144: move root column field to asset spec

### PR DESCRIPTION
**Pull request by ahaessly**

### CODE REVIEW

File: src/main/java/bio/terra/model/StudyJsonConversion.java [hunk 1]
```diff
@@ -148,31 +148,39 @@ public static AssetSpecification assetModelToAssetSpecification(AssetModel asset
...
+                colNamesToInclude = new ArrayList<>(allTableColumns.keySet());
+            }
+            Map<String, AssetColumn> assetColumnsMap = colNamesToInclude
                     .stream()
-                    .filter(entryToFilter -> tblMod.getColumns().isEmpty() ||
-                            tblMod.getColumns().contains(entryToFilter.getKey()))
-                    .map(entry -> new AssetColumn().setStudyColumn(entry.getValue()))
-                    .collect(Collectors.toList())));
-            newAssetTable.setColumns(assetColumns);
+                    .collect(Collectors.toMap(colName -> colName, colName ->
```
Comment (thathert) [inline src/main/java/bio/terra/model/StudyJsonConversion.java:181]: 👍 This pattern with `Collectors.toMap` is really cool

File: src/main/java/bio/terra/validation/StudyRequestValidator.java [hunk 2]
```diff
@@ -115,10 +115,24 @@ private void validateRelationship(RelationshipModel relationship, Errors errors,
         }
     }
 
-    private void validateAssetTable(AssetTableModel assetTable, Errors errors, SchemaValidationContext context) {
+    // returns true if the table is the root table
+    // error for invalid root column is determined within this method
+    private boolean validateAssetTable(
```
Comment (thathert) [inline src/main/java/bio/terra/validation/StudyRequestValidator.java:120]: nit: Can we update this name to something like `validateAssetTableAndCheckRoot` to distinguish between this and the other void validate methods?

File: src/test/java/bio/terra/controller/StudyTest.java [hunk 3]
```diff
@@ -289,11 +289,12 @@ public void testInvalidAssetTableColumn() throws Exception {
...
                 .name("participant")
-                .isRoot(true)
                 .columns(Collections.singletonList("date_collected"));
 
         AssetModel asset = new AssetModel()
                 .name("mismatched")
+                .rootTable("participant")
+                .rootColumn("date_collected")
                 .tables(Collections.singletonList(invalidAssetTable))
                 .follow(Collections.singletonList("participant_sample"));
```
Comment (thathert) [inline src/test/java/bio/terra/controller/StudyTest.java:300]: Can you add a test to catch the missing root column case?

**Review by thathert — CHANGES_REQUESTED** [review]
Looks good, a couple things requested.

**Review by RoriCremer — APPROVED** [review]

**Review by thathert — APPROVED** [review]

**Pull request merged**